### PR TITLE
New wrapper class

### PR DESCRIPTION
**Pull request by christianvedels**
This PR implements the basic prediction functionality of a new OccCANINE wrapper class. A lot of polishing still needed. But it is timely to agree on the overall structure now. 

Most important changes:
1. self.predict() is now itself just a wrapper of three specific other predict types: a) self._predict_flat(): Classical flat prediction b) self._predict_greedy(): Greedy prediction for seq2seq. c) self._predict_full(): Seq2seq probabilites for each of the 1919 potential HISCO codes.
2. Autodetection of model type based on features of loaded structure and numerous checks to ensure compatability between chosen prediction type, model type, and intended behavior
3. Entirely new Trie-based decoder for the full prediciton.

### CODE REVIEW

File: test_wrapper.py
Comment (TorbenSDJohansen) [inline test_wrapper.py:1]: I say we omit this from the PR (and just leave as an internal-only file, e.g. by adding it to .gitignore or just avoid committing it altogether).

Comment (christianvedels) [reply test_wrapper.py:1]: Agree. Updated .gitignore

**Comment by christianvedels** [comment]
Checked new commits by @TorbenSDJohansen. Looks good.

**Comment by TorbenSDJohansen** [comment]
Added suite of unit-tests (accidently not pushed in [d4c0591](https://github.com/christianvedels/OccCANINE/pull/114/commits/d4c0591d73435a63728600dc6b4b7e0214648533)), all of which pass -- but multiple of which pass due to "expected" `NotImplementedError` (with warnings raised). This is useful to keep track of any "outstanding" cases we may want to cover.

If you agree this appears to cover what we currently wants, I am ready to merge this into `dev` @christianvedels.

**Comment by christianvedels** [comment]
Great! With caveats noted above, I think this should be merged. If you agree, please go ahead.

**Pull request merged**